DockerfileLeanFoundation

```diff
@@ -53,193 +51,179 @@ RUN wget https://cdn.quantconnect.com/miniconda/${CONDA} && \
-RUN python -m nltk.downloader -d /usr/share/nltk_data punkt
-RUN python -m nltk.downloader -d /usr/share/nltk_data vader_lexicon
+RUN conda install -y -c conda-forge \
+    openmpi=4.0.3
 
-# Install TA-lib for python
-RUN wget http://cdn.quantconnect.com/ta-lib/ta-lib-0.4.0-src.tar.gz && \
-    tar -zxvf ta-lib-0.4.0-src.tar.gz && cd ta-lib && \
-    ./configure --prefix=/usr && make && make install && \
-    pip install TA-lib && cd .. && rm -irf ta-lib
+# Install non-math packages
+RUN conda install -y                \
+    blaze=0.11.3                    \
+    tensorflow-base=1.15.0
 
-# Install DX Analytics
-RUN wget https://cdn.quantconnect.com/dx/dx-master-9fab393.zip && \
-    unzip -q dx-master-9fab393 && cd dx-master && \
-    python setup.py install && cd .. && rm -irf dx-master
+# Install math/ML from pytorch
+RUN conda install -y -c pytorch     \
+    pytorch=1.5.0                   \
+    torchvision=0.6.0
 
-# Install py-earth
-RUN wget https://cdn.quantconnect.com/py-earth/py-earth-master-b209d19.zip && \
-    unzip -q py-earth-master-b209d19.zip && cd py-earth-master && \
-    python setup.py install && cd .. && rm -irf py-earth-master
+RUN conda clean -y --all
 
-# Install fastText
-RUN wget https://cdn.quantconnect.com/fastText/fastText-master-6d7c77c.zip && \
-    unzip -q fastText-master-6d7c77c.zip && cd fastText-master && \
-    python setup.py install && cd .. && rm -irf fastText-master
+RUN python -m nltk.downloader -d /usr/share/nltk_data punkt
+RUN python -m nltk.downloader -d /usr/share/nltk_data vader_lexicon
 
 # Update ODO
 RUN conda remove --force-remove -y odo
-RUN wget https://cdn.quantconnect.com/odo/odo-master-9fce669.zip && \
-    unzip -q odo-master-9fce669.zip && cd odo-master && \
-    python setup.py install && cd .. && rm -irf odo-master
+RUN pip install git+git://github.com/blaze/odo.git@9fce6690b3666160681833540de6c55e922de5eb
+
+# Install DX Analytics
+RUN pip install git+git://github.com/yhilpisch/dx.git@4c47c258acddf966be5082530764d8115831dcb3
 
 # Install Auto-KS
-RUN wget https://cdn.quantconnect.com/auto_ks/auto_ks-master-b39e8f3.zip && \
-    unzip -q auto_ks-master-b39e8f3.zip && cd auto_ks-master && \
-    python setup.py install && cd .. && rm -irf auto_ks-master
+RUN pip install git+git://github.com/cvxgrp/auto_ks.git@b39e8f39eec33d6c55a92e2a781d62a8439f32a6
 
 # Install Pyrb
-RUN wget https://cdn.quantconnect.com/pyrb/pyrb-master-d02b56a.zip && \
-    unzip -q pyrb-master-d02b56a.zip && cd pyrb-master && \
-    python setup.py install && cd .. && rm -irf pyrb-master
+RUN pip install git+git://github.com/jcrichard/pyrb.git@d02b56ae91f67b73f39e5e474fa90b0523f6638b
 
 # Install SSM
-RUN wget https://cdn.quantconnect.com/ssm/ssm-master-34b50d4.zip && \
-    unzip -q ssm-master-34b50d4.zip && cd ssm-master && \
-    python setup.py install && cd .. && rm -irf ssm-master
+RUN pip install git+git://github.com/slinderman/ssm.git@9fd66aeddf74059c934c8b214f8f7757b9bef214
```

Comment:
Curious why we replaced `cdn.quantconnect` for specific commits?

Reply:
Since some packages host the version in GitHub, we are using those in #4465. E.g.:
> https://github.com/jakobrunge/tigramite/archive/4.1.zip

and the fixed hash when there is no archive.
However, since a GitHub repo admin can wipe out the commit history like `mlfinlab` did, maybe we should keep them in `cdn.quantconnect`.